### DISCUSSION TITLE
The extensions of dialog and the path of dialog

### DISCUSSION THREAD

**Discussion by xueweiwujxw** [tauri-apps/tauri#5712] [Q&A] [answered]
Hello, I wanna open a dialog to select a file with any extensions.
```typescript
let select = await open({
    title: title,
    multiple: false,
    directory: false,
    defaultPath: await localDataDir(),
    filters: [
      { name: '*', extensions: ['*'] },
    ],
  });
```
I set the filters as `*` extensions. However, I can not find the file I want.
And I  want to open a folder that is the latest path, what is the value of `defaultPath` I should set?
Please help us. Thanks a lot.

**Comment by FabianLars** [comment, marked as answer]
I'd suggest just removing the `filters` property all together, this way it really should list all files.

> And I want to open a folder that is the latest path

You'd have to keep track of it yourself. Like store the last used path _somewhere_ and then set it as the defaultPath dynamically (remoing the file itself from the path, it should point to the folder)

**Reply by xueweiwujxw** [reply]
Thanks. And are there any suggestions to store the last path? Using rust to store the path and listening to the events of the front end?

**Reply by FabianLars** [reply]
Yeah that could work, sessionstorage/localstorage probably works too.

**Reply by xueweiwujxw** [reply]
Yeah. I use SQLite to save the last path for persistent storage
